src/main/java/com/redhat/devtools/lsp4ij/LanguageServerWrapper.java

```diff
@@ -366,16 +370,32 @@ public synchronized void start() throws LanguageServerException {
                             initParams.setRootPath(rootURI.getPath());
                         }
 
+                        // Get the useIntegerIds setting for this language server
+                        boolean useIntegerIds = false;
+                        if (serverDefinition instanceof ClientConfigurableLanguageServerDefinition clientConfigurableDefinition) {
```

Comment:
Thanks for having switched to client configuration. The next step now id to manage any language server definition: user-defined and from extension plugin (ls  embedded in an IJ plugin).

The main idea is to not use ClientConfigurableLanguageServerDefinition which is specific for user defined ls, but use LSPClieantFeatures. In other words you should write a code like this (perhaps the method name is not very good)

`boolean useIntegerIds  =  getClientFeatures().isUseJsonRpcIntegerIds()`

To implement this method isUseJsonRpcIntegerIds, I suggets that you see LSPCLientFeatures#getBlock#getCommentPrefix and impl in UserDefinedClientFeatures#getCommentPrefix